sdk/aqueduct/backend/api_client.py

```diff
@@ -272,6 +275,86 @@ def connect_integration(
         resp = requests.post(url, url, headers=headers)
         self.raise_errors(resp)
 
+    def get_dag_engine_status(
+        self,
+        dag: DAG,
+    ) -> Dict[str, EngineStatusResponse]:
+        """Makes a request against the /api/integration/engine/status endpoint.
+
+        Args:
+            dag:
+                The DAG object. We will extract the engine integration IDs and send them
+                to the backend to retrieve their status. Currently, we are only interested in
+                the status of dynamic engines.
+
+        Returns:
+            A EngineStatusResponse object, parsed from the backend endpoint's response.
+        """
+        engine_integration_ids = set()
+
+        dag_engine_config = dag.engine_config
+        if dag_engine_config.type == RuntimeType.K8S:
+            assert dag_engine_config.k8s_config is not None
+            engine_integration_ids.add(str(dag_engine_config.k8s_config.integration_id))
+        for op in dag.operators.values():
+            if op.spec.engine_config and op.spec.engine_config.type == RuntimeType.K8S:
+                assert op.spec.engine_config.k8s_config is not None
+                engine_integration_ids.add(str(op.spec.engine_config.k8s_config.integration_id))
+
+        return self.get_engine_status(list(engine_integration_ids))
+
+    def get_engine_status(
+        self,
+        engine_integration_ids: List[str],
+    ) -> Dict[str, EngineStatusResponse]:
+        """Makes a request against the /api/integration/engine/status endpoint.
+
+        Args:
+            engine_integration_ids:
+                A list of engine integration IDs. Currently, we are only interested in
+                the status of dynamic engines.
```

Comment:
What happens if a non-valid engine is supplied (not K8s or dynamic)? Should we error here or at least document that assumption.